### PR TITLE
Sample implementation of embedding theme files

### PR DESCRIPTION
**Pull request by LordFlashmeow**
Building on #25, here's a sample of how we can embed the theme and filetypes files into the executable.

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you very much for your contribution!

To be honest, I'd like to avoid doing this. As far as I can see, the only reason would be the packaging difficulties on macOS. I can't really believe that we can not solve this in another way. It should be possible to install a few files to a specified location and load them from there.

**Comment by LordFlashmeow** [comment]
It's not just packaging difficulties with macOS, but packaging difficulties on every platform. Currently, unless you are using debian or Arch, you need to download the repo and copy the files by hand. As I've discovered, every system is different, and the less we depend on the system, the better.

Cargo doesn't let us copy files to a directory when doing `cargo install`, we need a full-on package manager to handle that for us instead.

If we embed the files, then it is possible for the average user to download the binary from the releases page and it'll work no matter where they put it. Users can also use `cargo install vivid` without any further action.

The only time a user would need to add a file to the config directory would be if they wanted to customize one of the themes, and even then they also have the option of specifying the exact file via the command line or an environment variable.

Embedding allows:
* Portable executable
* No setup for majority of users
* Minimal change in binary size (less than .1 MB larger)
* `cargo update` adds new themes when pushed to repo

**Comment by sharkdp** [comment]
You have some excellent arguments, thank you. I will give this a review.

**Comment by sharkdp** [comment]
> Rust Embed
> Rust Custom Derive Macro which loads files into the rust binary at compile time during release and loads the file from the fs during dev.

Oh, that's pretty cool!

File: src/main.rs [hunk 1]
```diff
@@ -47,9 +52,16 @@ fn load_filetypes_database(matches: &ArgMatches, user_config_path: &PathBuf) ->
 
     let database_path = database_path_from_arg
         .or(database_path_env)
-        .or_else(|| util::get_first_existing_path(&[&database_path_user, database_path_system]))
-        .ok_or(VividError::CouldNotFindDatabase)?;
```
Comment (sharkdp) [inline src/main.rs:51]: Can we remove `VividError::CouldNotFindDatabase` in that case?
```
warning: variant is never constructed: `CouldNotFindDatabase`
  --> src/error.rs:15:5
   |
15 |     CouldNotFindDatabase,
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

```

Comment (LordFlashmeow) [reply src/main.rs:51]: I think so, although we will need to keep the CouldNotFindTheme error

**Comment by sharkdp** [comment]
The code changes look great - thank you very much!

**Comment by LordFlashmeow** [comment]
```
    let theme = sub_matches
        .value_of("theme")
        .or_else(|| theme_from_env.as_ref().map(String::as_str))
        .unwrap_or("molokai");
```

In this bit, it looks like we're setting the theme to molokai when unwrapping the environment theme fails. Can we just raise an error in that case?

**Comment by sharkdp** [comment]
> In this bit, it looks like we're setting the theme to molokai when unwrapping the environment theme fails. Can we just raise an error in that case?

Yes - sounds good. Thanks

**Comment by LordFlashmeow** [comment]
Turns out it's checking if no theme is specified. Should that raise an error?

**Comment by sharkdp** [comment]
Well, for now, `vivid generate` will use molokai as a default if `VIVID_THEME` is not set. I'm okay with changing this.

Ideally, we would cover all cases:
* `VIVID_THEME` unset => `vivid generate` should throw an error.
` unknown `VIVID_THEME` => `vivid generate` should throw an error.
* `vivid generate <theme>` for unknown `<theme>` => should throw an error.

**Comment by sharkdp** [comment]
Thank you for the update!

**Pull request merged**